sdk/include/opentelemetry/sdk/common/attributemap_hash.h

```diff
@@ -71,7 +71,7 @@ inline size_t GetHashForAttributeMap(
         {
           return true;
         }
-        GetHash(seed, key.data());
```

Comment:
In the `GetHash` function, could we assert that `arg` should not be a pointer?

Reply:
We can add it, but since this is an internal API, it might not be necessary as long as we ensure it's used correctly

Reply:
Thinking again, we do need to support the `GetHash` function with char pointer, as the [AttributeValue ](https://github.com/open-telemetry/opentelemetry-cpp/blob/6dbfdb52ebca82d8705392fe88dadd9c133374e4/api/include/opentelemetry/common/attribute_value.h#L37)enum contains it.

Reply:
If we do need supporter pointer, perhaps create a dedicated specialization for it? Then a difference hash function could be applied to pointers.

Reply:
yes, it is done in the latest commit.